src/v/cluster/controller_backend.cc

```diff
@@ -473,6 +478,40 @@ ss::future<> controller_backend::fetch_deltas() {
       });
 }
 
+ss::future<> controller_backend::clear_orphan_topic_files() {
+    auto topics = _topics.local().all_topics();
+    storage::log_manager::ntp_to_revision topics_on_node;
```

Comment:
q: This is a map of ntp -> revision of all the ntps in the system, can be potentially big. What is the use of constructing this up front? When recursing through the topic/partition directories can't we just lookup the topic table? Something like..

```
ntp foo = partition_path::parse_partition_directory()

if (foo.revision_id < last_applied_id_snapshot && !topic_table.contains() && !topic_table.previous_replicas.contains())  {
  cleanup();
}

```

Just trying to understand the use of creating a snapshot of all the ntps in the system.

Reply:
Previously I was thinking about creating this map to get snapshot of topictable right after bootstrap.
But I agree that we don't need it now

Reply:
+1 on using the topic table directly.